models/src/anemoi/models/preprocessing/postprocessor.py

```diff
@@ -0,0 +1,354 @@
+                f"({self.num_training_output_vars}) or inference shape ({self.num_inference_output_vars})",
+            )
+
+        # Replace values
+        for postprocessor, idx_dst in zip(self.postprocessorfunctions, index):
+            if idx_dst is not None:
+                x[..., idx_dst] = postprocessor(x[..., idx_dst])
+        return x
+
+
+class CustomRelu(torch.nn.Module):
+    """Custom ReLU activation function with a specified threshold."""
+
+    def __init__(self, threshold: float = 0.0) -> None:
+        """Initialize the CustomReLU with a specified threshold.
+
+        Parameters
+        ----------
+        threshold : float
+            The threshold for the ReLU activation.
+        """
+        super().__init__()
+        self.threshold = threshold
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        """Apply the ReLU activation with the specified threshold.
+
+        Parameters
+        ----------
+        x : torch.Tensor
+            The input tensor to process.
+
+        Returns
+        -------
+        torch.Tensor
+            The processed tensor with ReLU applied.
+        """
+        return torch.nn.functional.relu(x - self.threshold) + self.threshold
+
+
+class NormalizedReluPostprocessor(Postprocessor):
+    """Postprocess with a ReLU activation and customizable thresholds.
+
+    Expects the config to have keys corresponding to customizable thresholds as lists of variables to impute.:
+    ```
+    nornmalizer: 'mean-std'
+    1:
+        - y
+    0:
+        - x
+    3.14:
+        - q
+    ```
+    Thresholds are in un-normalized space. If normalizer is specified, the threshold values are not normalized.
+    This is necessary if in config file the normalizer is specified before the postprocessor, e.g.:
+    ```
+    data:
+        processors:
+          normalizer:
+            _target_: anemoi.models.preprocessing.normalizer.InputNormalizer
+            config:
+              default: "mean-std"
+          normalized_relu_postprocessor:
+            _target_: anemoi.models.preprocessing.postprocessor.NormalizedReluPostprocessor
+            config:
+              271.15:
+              - x1
+              0:
+              - x2
+            normalizer: 'mean-std'
+    """
+
+    def __init__(
+        self,
+        config=None,
+        data_indices: Optional[IndexCollection] = None,
+        statistics: Optional[dict] = None,
+        normalizer: str = "none",
+    ) -> None:
+
+        self.normalizer = normalizer
+        self.statistics = statistics
+
+        # Validate normalizer input
+        if self.normalizer not in {"none", "mean-std", "min-max", "max", "std"}:
+            raise ValueError(
+                "Normalizer must be one of: 'none', 'mean-std', 'min-max', 'max', 'std' in NormalizedReluBounding."
+            )
+
+        super().__init__(config, data_indices, statistics)
+
+    def _create_imputation_indices(
```

Comment:
Why does this function need to be redefined for each post-processor new class? I think there should be a way to abstract away some of this complexity into the base `Postprocessor` and then call the super method to avoid repeating code